paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -358,8 +396,45 @@ class ConvMKLDNNGradOpKernel : public paddle::framework::OpKernel<T> {
     std::vector<int> src_tz = paddle::framework::vectorize2int(input->dims());
     std::vector<int> weights_tz =
         paddle::framework::vectorize2int(filter->dims());
+    int g = std::max(groups, 1);
+    if (g > 1) {
+      if (is_conv3d) {
+        int o = weights_tz[0];
+        int i = weights_tz[1];
+        int d = weights_tz[2];
+        int h = weights_tz[3];
+        int w = weights_tz[4];
+        weights_tz.resize(6);
+        weights_tz[0] = g;
+        weights_tz[1] = o / g;
+        weights_tz[2] = i;
+        weights_tz[3] = d;
+        weights_tz[4] = h;
+        weights_tz[5] = w;
+      } else {
+        int o = weights_tz[0];
+        int i = weights_tz[1];
+        int h = weights_tz[2];
+        int w = weights_tz[3];
+        weights_tz.resize(5);
+        weights_tz[0] = g;
+        weights_tz[1] = o / g;
+        weights_tz[2] = i;
+        weights_tz[3] = h;
+        weights_tz[4] = w;
+      }
+    }
```

Comment:
line 400-426 are same with line 91-115. please make a function and reuse it.

Reply:
Done